### PR TITLE
Define unified header semantics across all transports

### PR DESCRIPTION
**Pull request by vasilvv**

### CODE REVIEW

File: draft-ietf-webtrans-overview.md [hunk 1]
```diff
@@ -218,20 +218,49 @@ Policy [CSP].
...
-instance, in protocols creating a new TLS 1.3 session {{!RFC8446}}, this would
-mean that the user agent MUST NOT treat the session as established until it
-received a Finished message from the server.
+WebTransport session begins with a handshake in which the client and the server
+exchange a set of header fields.  This provides an opportunity for the parties
+to exchange the metadata such as the origin of the client and the path
+component of the URL.
+
+A WebTransport session MUST NOT be considered established until it is secure
+against replay attacks.  For instance, in protocols creating a new TLS 1.3
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-overview.md:227]: nit: I'd say QUIC or TLS 1.3

File: draft-ietf-webtrans-overview.md [hunk 2]
```diff
@@ -218,20 +218,49 @@ Policy [CSP].
...
 
 In some cases, the transport protocol might allow transmitting data before the
 session is established; an example is TLS 0-RTT data.  Since this data can be
 replayed by attackers, it MUST NOT be used unless the client has explicitly
 requested 0-RTT for specific streams or datagrams it knows to be safely
 replayable.
 
+## Headers
+
+Any WebTransport protocol MUST provide a method for peers to exchange headers.
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-overview.md:239]: I would move this to after the definition of headers

File: draft-ietf-webtrans-overview.md [hunk 3]
```diff
@@ -218,20 +218,49 @@ Policy [CSP].
...
+The client sends its header fields first; the server sends its headers in
+response to the ones received from the client.
+
+WebTransport headers are an unordered set of key-value pairs referred to as
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-overview.md:243]: I'd stick to "key" or "name" instead of using both?

**Review by DavidSchinazi — APPROVED** [review]

**Comment by LPardue** [comment]
I asked this question during the IETF 109 session but due to the technical difficulties I said I'd put it on this ticket.

@vasilvv, do you believe that this change is a good one even if the group were to to decide that no-one wants the HTTP-based WebTransports?

**Comment by LPardue** [comment]
As a general observation about this change, the header rules seem quite loose. In HTTP, we send messages. But this change talks about header fields and things that look a bit like requests or responses but aren't. I'm not really sure how I would write a server-side WebTransport parser with confidence, especially if I already have code that handles HTTP/3 framing and messages.

**Pull request closed without merging**